src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,30 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CHAR = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can't be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with such login is already in the storage");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < MIN_CHAR
+                || user.getLogin().isBlank()) {
```

Comment:
isBlank method is excessive, cannot be true if user.getLogin().length() is fine
```suggestion
        if (user.getLogin() == null || user.getLogin().length() < MIN_CHAR) {
```